### PR TITLE
Document Pose::operator*()

### PR DESCRIPTION
**Pull request by sloretz**
Adds some documentation for `Pose::operator*()`. If I understand correctly, it looks like `B + A` is the same as `A * B`, but [operator+ is discouraged](https://github.com/ignitionrobotics/ign-math/issues/60).

### CODE REVIEW

**Comment by chapulina** [comment]
How about adding a couple of illustrative test cases? I don't see that operator being explicitly tested here:

https://github.com/ignitionrobotics/ign-math/blob/main/src/Pose_TEST.cc

Maybe it's tested indirectly from somewhere else.

**Comment by scpeters** [comment]
we added tests for `Pose*` in `Matrix4_TEST` when we changed its behavior to resolve #60:

* https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-math/pull-requests/301/page/1
* https://github.com/ignitionrobotics/ign-math/compare/9552cd5ce7eaed6651b726fc03a23b637db83675..7b51f5610930caa03966cf41bf2014bf547cd749#diff-264517f0fd7cdc07fe41f4522b6a18fb832db7215be636d33eae8cb7084ae8e4

**Comment by scpeters** [comment]
for the life of me, I have the toughest time verbalizing how these transforms work; I'm sure @azeey remembers how much trouble I had when we were writing the following SDFormat tutorial:

* http://sdformat.org/tutorials?tut=specify_pose&cat=specification&

I'll let him speak up if he has any concrete suggestions for syntax

**Comment by azeey** [comment]
Thanks for adding the documentation @sloretz. In most robotics books (eg. http://hades.mech.northwestern.edu/images/2/25/MR-v2.pdf), the `*` operator is used for composing transforms rather than `+`. The notation given in these books uses two frames to describe a transform/pose. In http://sdformat.org/tutorials?tut=specify_pose&cat=specification&, we have used `X_BA` to indicate the transform from B to A. This is also the same quantity as the pose of frame A (relative to and expressed in) frame B. 

Given two `Pose` objects, the order of arguments when using `Pose::operator*` is the same as if the `Pose`s were first converted to 4x4 homogeneous matrices and the product of the matrices was taken. So if we have frames `O`, `P`, and `Q` and we have the poses `X_OP` (frame `P` relative to `O`) and `X_PQ` (frame `Q` relative to `P`), then `X_OQ` (frame `Q` relative to `O`) would be `X_OQ = X_OP * X_PQ`

**Comment by sloretz** [comment]
> Given two Pose objects, the order of arguments when using Pose::operator* is the same as if the Poses were first converted to 4x4 homogeneous matrices and the product of the matrices was taken. So if we have frames O, P, and Q and we have the poses X_OP (frame P relative to O) and X_PQ (frame Q relative to P), then X_OQ (frame Q relative to O) would be X_OQ = X_OP * X_PQ

Uses this notation in 975a50c

**Review by azeey — APPROVED** [review]
LGTM

**Comment by sloretz** [comment]
Does this need a second reviewer, or is it good to merge? If the latter, would someone mind merging it? I don't have write access to the ignitionrobotics repos.

**Review by chapulina — APPROVED** [review]
:ok_hand:

**Pull request merged**
